src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -703,11 +709,13 @@ public static IEnumerable<TSource> SkipWhile<TSource>(this IEnumerable<TSource>
 
         private static IEnumerable<TSource> SkipWhileIterator<TSource>(IEnumerable<TSource> source, Func<TSource, bool> predicate)
         {
-            bool yielding = false;
-            foreach (TSource element in source)
+            using (IEnumerator<TSource> e = source.GetEnumerator())
```

Comment:
It hardly sounded like that. If you would like my to be less profane, please be more clear in what you mean. Fair?